pkg/diffs/diffs.go

```diff
@@ -0,0 +1,136 @@
+package diffs
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/api/equality"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+// PreSubmits holds the logic of prow's presubmits yaml format.
+type PreSubmits struct {
+	PreSubmits map[string][]prowconfig.Presubmit `yaml:"presubmits,omitempty"`
+}
+
+// Diffs holds the logic about the differences between the files.
+type Diffs struct {
+	JobConfigPath  string
+	CiOpConfigPath string
+	DecorationPath string
+	ChangedFiles   []string
+}
+
+// NewDiffs creates a new Diffs{}
+func NewDiffs(jobConfigPath, ciOpConfigPath, decorationPath string) *Diffs {
+	return &Diffs{
+		JobConfigPath:  jobConfigPath,
+		CiOpConfigPath: ciOpConfigPath,
+		DecorationPath: decorationPath,
+	}
+
+}
+
+func (d *Diffs) findChangedFiles() error {
+	// Walk through the cloned repo
+	if err := filepath.Walk(fmt.Sprintf("%s/ci-operator/jobs/", d.DecorationPath), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			log.Printf("prevent panic by handling failure accessing a path %q: %v\n", path, err)
+			return err
+		}
+
+		if !info.IsDir() && filepath.Ext(path) == ".yaml" {
+			masterFileContents, err := ioutil.ReadFile(fmt.Sprintf("%s/%s", d.JobConfigPath, filepath.Base(path)))
+			if err != nil {
+				log.Printf("Can't read master's file: %v", err)
+			}
+			prFileContents, err := ioutil.ReadFile(path)
+			if err != nil {
+				log.Printf("Error reading file %s: %v", path, err)
+			}
+			if bytes.Compare(masterFileContents, prFileContents) != 0 {
+				d.ChangedFiles = append(d.ChangedFiles, path)
+			}
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Diffs) getPresubmitChanges() []string {
+	var preSubmitChangedFiles []string
+	if err := d.findChangedFiles(); err != nil {
+		log.Fatal("Couldn't populate changed files.")
```

Comment:
This should not `Fatal`, but pass the error out like everything else.

Reply:
If that fails, there is no reason to continue.